src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1185,12 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if App gives permission to AccountManager
+            if(!verifyManifestPermissions())
+            {
+                return result;
```

Comment:
Shouldn't we return back the exception?
we should do like:
<code>
callbackHandler.onError(exception);
return null;
</code>